integration/mediation-tests/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/servers/GreenMailServer.java

```diff
@@ -0,0 +1,237 @@
+        SearchTerm searchTerm = new AndTerm(new SubjectTerm(emailSubject), new BodyTerm(emailSubject));
+        Message[] messages = mailFolder.search(searchTerm);
+        for (Message message : messages) {
+            if (message.getSubject().contains(emailSubject)) {
+                log.info("Found the Email with Subject : " + emailSubject);
+                emailReceived = true;
+                break;
+            }
+        }
+        if (store != null) {
+            try {
+                store.close();
+            } catch (MessagingException e) {
+                log.warn("Error when closing the store ", e);
+            }
+        }
+        return emailReceived;
+    }
+
+    /**
+     * Check inbox and make sure a particular email is deleted
+     *
+     * @param emailSubject
+     * @param protocol
+     * @return
+     * @throws ESBMailTransportIntegrationTestException
+     */
+    public static boolean checkEmailDeleted(String emailSubject, String protocol)
+            throws ESBMailTransportIntegrationTestException, MessagingException {
+        boolean isEmailDeleted = false;
+        long startTime = System.currentTimeMillis();
+
+        while ((System.currentTimeMillis() - startTime) < WAIT_TIME_MS) {
+            if (!isMailReceivedBySubject(emailSubject, EMAIL_INBOX, protocol)) {
+                log.info("Email has been deleted successfully!");
+                isEmailDeleted = true;
+                break;
+            }
+            try {
+                Thread.sleep(500);
+            } catch (InterruptedException e) {
+                log.warn("Error during thread sleep for 0.5 sec");
+            }
+        }
+        return isEmailDeleted;
+    }
+
+    /**
+     * Check trash and make sure a particular email is moved to trash
+     *
+     * @param emailSubject
+     * @param protocol
+     * @return
+     * @throws ESBMailTransportIntegrationTestException
+     */
+    public static boolean CheckEmailMoved(String emailSubject, String protocol)
+            throws ESBMailTransportIntegrationTestException, MessagingException {
+        boolean mailReceived = false;
+        long startTime = System.currentTimeMillis();
+        while ((System.currentTimeMillis() - startTime) < WAIT_TIME_MS) {
+            if (isMailReceivedBySubject(emailSubject, EMAIL_TRASH, protocol)) {
+                log.info("Found the expected email in mailbox : " + emailSubject);
+                mailReceived = true;
+                break;
+            }
+            try {
+                Thread.sleep(500);
+            } catch (InterruptedException e) {
+                log.warn("Error during thread sleep for 0.5 sec");
+            }
+        }
+        return mailReceived;
+    }
+
+    /**
+     * Delete all unread emails in the inbox
+     *
+     * @param protocol
+     * @throws ESBMailTransportIntegrationTestException
+     */
+    public static void deleteAllUnreadEmails(String protocol)
```

Comment:
shall we add another method to delete all mails regardless of read/unread

Reply:
done